app/views/my/solutions/solve.html.haml

```diff
@@ -0,0 +1,105 @@
+.lo-container
+  #solve-interface
+    #ace-editor
+      :preserve
+        class TwoFer
+          def self.two_fer(name="you")
+            "One for %s, one for me." % name
+          end
+        end
+
+    #tests-output
+      .ops-status
+        Ops Status:
+        %span
+      .results-status
+        Tests Status:
+        %span
+      .results-message
+        Message:
+        %span
+      .test-failure
+        Failed test:
+        %span.name
+        .message
+
+  %button#run-tests-button.pure-button Run Tests
+
+:scss
+  #solve-interface {
+    display:flex;
+  }
+
+  #ace-editor {
+    width:50%;
+    height:300px;
+    margin-top:10px;
+    margin-bottom:10px;
+  }
+  #tests-output {
+    padding-left:50px;
+
+    .ops-status,
+    .results-status,
+    .results-message,
+    .test-failure {
+      display:none;
+    }
+  }
+  #run-tests-button {
+    background:green;
+  }
+
+-content_for :js do
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.4.6/ace.js" type="text/javascript" charset="utf-8"></script>
+  :javascript
+    // Set interval tracker for use later
+    // Use an array to avoid races
+    var intervals = {}
+
+    var editor = ace.edit("ace-editor");
+    editor.setTheme("ace/theme/monokai");
+    editor.session.setMode("ace/mode/#{@track.slug}");
+
+    $('#run-tests-button').click(function() {
+      var code = editor.getValue();
+      $.ajax(
+        "/my/solutions/#{@solution.uuid}/submit",
+        {
+          method: "PATCH",
+          data: {
+            files: {
+              "two_fer.rb": code
+            }
+          }
+        }
+      ).done(function(data) {
+        submissionId = data.submission.id
+        $('#tests-output .ops-status').show().children('span').html("Tests pending")
```

Comment:
When you see the following pattern:

```javascript
$("parent child")
$("parent child")
$("parent child")
```

You probably should memoize it, so that when you _change_ the selector, you don't need Shotgun Surgery:

```javascript
var $opsStatus = $("#tests-output .ops-status")
$opsStatus.show().children('span')...
```

Reply:
When you see the following pattern:

```javascript
$("parent child1")
$("parent child2")
$("parent child3")
```

it can make sense to extract parent first. This automatically indicates a (sub)-component:

```javascript
var $testsOutput = $('#tests-output')

var $opsStatus = $testsOutput.find('.ops-status')
var $opsStatus = $testsOutput.find('.results-status')
var $opsStatus = $testsOutput.find('.results-message')
```

Reply:
Right now you're using `classes` to determine where things go. That's _okay_, but not great. Classes can be re-used, and you're mixing IDS with classes, but you _always_ only select the first result from the query. This means that if I add a second `.ops-status` on the page, it might work, breaking the old one, or might not work, preservering the old one.

This probably indicates that you want scoping. The scope is not an HTML scope, but a "data" scope. It's the ops-status of the tests you're running. So the scope is probably "the current iteration of the current exercise in the current track".